framework/http/FileStream.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\http;
+
+use Psr\Http\Message\StreamInterface;
+use Yii;
+use yii\base\BaseObject;
+use yii\base\ErrorHandler;
+use yii\base\InvalidConfigException;
+
+/**
+ * FileStream represents file stream.
+ *
+ * Example:
+ *
+ * ```php
+ * $stream = new FileSteam([
+ *     'filename' => '@app/files/items.txt',
+ *     'mode' => 'w+',
+ * ]);
+ *
+ * $stream->write('some content...');
+ * $stream->close();
+ * ```
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.1.0
+ */
+class FileStream extends BaseObject implements StreamInterface
```

Comment:
There's no benefit in extending from `BaseObject` in this case.

Reply:
Same as [here](https://github.com/yiisoft/yii2/pull/14701#discussion_r135021442)

Reply:
We should finally extract it into to traits :( Both aren't good enough reasons to keep inheritance tree as big as it is now.